src/Microsoft.ML.Fairlearn/metrics/FairlearnMetricCatalog.cs

```diff
@@ -0,0 +1,291 @@
+
+            // create the dictionary to hold the results
+            Dictionary<string, double> metricsDict = new Dictionary<string, double>();
+            metricsDict.Add("AUC", metrics.AreaUnderRocCurve);
+            metricsDict.Add("Accuracy", metrics.Accuracy);
+            metricsDict.Add("PosPrec", metrics.PositivePrecision);
+            metricsDict.Add("PosRecall", metrics.PositiveRecall);
+            metricsDict.Add("NegPrec", metrics.NegativePrecision);
+            metricsDict.Add("NegRecall", metrics.NegativeRecall);
+            metricsDict.Add("F1Score", metrics.F1Score);
+            metricsDict.Add("AreaUnderPrecisionRecallCurve", metrics.AreaUnderPrecisionRecallCurve);
+            // following metrics are from the extensions
+            metricsDict.Add("LogLoss", metrics.LogLoss);
+            metricsDict.Add("LogLossReduction", metrics.LogLossReduction);
+            metricsDict.Add("Entropy", metrics.Entropy);
+            return metricsDict;
+        }
+    }
+    public class RegressionGroupMetric : IGroupMetric
+    {
+        private readonly IDataView _eval;
+        private readonly string _labelColumn;
+        private readonly string _scoreColumn;
+        private readonly string _sensitiveFeatureColumn;
+        private readonly MLContext _context = new MLContext();
+
+        public RegressionGroupMetric(IDataView eval, string labelColumn, string scoreColumn, string sensitiveFeatureColumn)
+        {
+            _eval = eval;
+            _labelColumn = labelColumn;
+            _scoreColumn = scoreColumn;
+            _sensitiveFeatureColumn = sensitiveFeatureColumn;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public DataFrame ByGroup()
+        {
+            // 1. group row according to sensitive feature column
+            // 2. split dataset to different groups, data_g1, data_g2.....
+            // 3. calculate binary metrics for different groups
+            // 4. create datafrome from result of step 3
+            // 5. return it.
+            var sensitiveCol = _eval.Schema[_sensitiveFeatureColumn];
+            // get all the columns of the schema
+            DataViewSchema columns = _eval.Schema;
+
+            // TODO: is converting IDataview to DataFrame the best practice?
+            // .ToDataFram pulls the data into memory.
+
+            //Brainstorm:  1. save it to a text file, temp file. figure unique columns. do a filter on those columns
+            // 2. filtering (maybe not the best approach) dataview
+            // 3. custom mapping 
+            var evalDf = _eval.ToDataFrame();
+            var groups = evalDf.Rows.GroupBy(r => r[sensitiveCol.Index]);
+            var groupMetric = new Dictionary<object, RegressionMetrics>();
+            foreach (var kv in groups)
```

Comment:
It looks like most of the logic in the loop is identical to the BinaryGroupMetric. Is there a way you could reuse the code instead of just duplicating it?